web/src/activity.ts

```diff
@@ -191,16 +193,54 @@ export function mark_client_active(): void {
     }
 }
 
+export function setup_browser_idle_detector(idle_handler: {cancel: () => void}): void {
+    // This code is separated out of `initialize` to facilitate testing
+    if (!browser_idle_detection.is_browser_idle_detector_supported()) {
+        blueslip.log("Browser idle detector not supported");
+        return;
+    }
+
+    $(document).one("keypress click", () => {
+        void browser_idle_detection.request_browser_idle_detector_permission();
+    });
+
+    browser_idle_detection.listen_to_idle_detector_permission_change(() => {
```

Comment:
I don't particularly like the redundant method names; they could be considerably shorter. For example, `browser_idle_detection.add_permission_change_listener();` is much more Javascripty (ala `addEventListener`).